### PR TITLE
feat(route): add route of 深圳福田园岭小学

### PR DESCRIPTION
**Pull request by valuex**
## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例

```routes
/szftedu/gonggao
/szftedu/dongtai
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [X ] New Route / 新的路由
  - [ ] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [ ] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [ ] Parsed / 可以解析
  - [ ] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

### CODE REVIEW

File: lib/routes/szftedu/gonggao.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    name: '园岭小学公告',
+    maintainers: ['Valuex'],
+    handler,
+    description: ``,
+};
+
+async function handler(ctx) {
```
[the remark this answers is not in the document]
Comment (valuex) [reply lib/routes/szftedu/gonggao.ts:30]: This error can be ignored. If possbile, please define it as a warning, not an error.

File: lib/routes/szftedu/dongtai.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    name: '园岭小学动态',
+    maintainers: ['Valuex'],
+    handler,
+    description: ``,
+};
+
+async function handler(ctx) {
```
[the remark this answers is not in the document]
Comment (valuex) [reply lib/routes/szftedu/dongtai.ts:29]: This error can be ignored. If possbile, please define it as a warning, not an error.

**Review by TonyRL — COMMENTED** [review]
Please split PR for different namespaces into separated PRs.

**Comment by valuex** [comment]
> Please split PR for different namespaces into separated PRs.

Thanks for your comment. Will split it.
And I have a question, because of merging into the main branch take some time for every single namespace, how can contributor to make serveral PRs at the same time?

File: lib/routes/szftedu/dongtai.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    },
+    name: '园岭小学动态',
+    maintainers: ['Valuex'],
+    handler,
+    description: ``,
+};
+
+async function handler() {
+    const link = host;
+    const base_link = `https://ylxx.szftedu.cn`;
```
Comment (TonyRL) [inline lib/routes/szftedu/dongtai.ts:31]:
```suggestion
    const base_link = 'https://ylxx.szftedu.cn';
```
`camelCase`

File: lib/routes/szftedu/dongtai.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    features: {
+        requireConfig: false,
+        requirePuppeteer: false,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    name: '园岭小学动态',
+    maintainers: ['Valuex'],
```
Comment (TonyRL) [inline lib/routes/szftedu/dongtai.ts:24]:
```suggestion
    maintainers: ['valuex'],
```

File: lib/routes/szftedu/dongtai.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    parameters: {},
+    features: {
+        requireConfig: false,
+        requirePuppeteer: false,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    name: '园岭小学动态',
```
Comment (TonyRL) [inline lib/routes/szftedu/dongtai.ts:23]: No need to repeat namespace name in route name.

File: lib/routes/szftedu/dongtai.ts [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+        }));
+
+    const items = await Promise.all(
+        lists.map((item) =>
+            cache.tryGet(item.link, async () => {
+                const thisUrl = item.link;
+                const trueLink = thisUrl.includes("http") ? thisUrl : base_link + thisUrl;
+                const response = await got.get(trueLink);
+                const $ = load(response.data);
+                item.description = $('body').html();
```
Comment (TonyRL) [inline lib/routes/szftedu/dongtai.ts:52]: Do not return the whole HTML body.

File: lib/routes/szftedu/gonggao.ts [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    parameters: {},
+    features: {
+        requireConfig: false,
+        requirePuppeteer: false,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    name: '园岭小学公告',
```
Comment (TonyRL) [inline lib/routes/szftedu/gonggao.ts:24]: No need to repeat namespace name in route name.

File: lib/routes/szftedu/gonggao.ts [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+    maintainers: ['Valuex'],
```
Comment (TonyRL) [inline lib/routes/szftedu/gonggao.ts:25]:
```suggestion
    maintainers: ['valuex'],
```

File: lib/routes/szftedu/gonggao.ts [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+        }));
+
+    const items = await Promise.all(
+        lists.map((item) =>
+            cache.tryGet(item.link, async () => {
+                const thisUrl = item.link;
+                const trueLink = thisUrl.includes("http") ? thisUrl : host + thisUrl.substring(1);
+                const response = await got.get(trueLink);
+                const $ = load(response.data);
+                item.description = $('body').html();
```
Comment (TonyRL) [inline lib/routes/szftedu/gonggao.ts:50]: Do not return the whole HTML body.

File: lib/routes/szftedu/namespace.ts [hunk 10]
```diff
@@ -0,0 +1,6 @@
+import type { Namespace } from '@/types';
+
+export const namespace: Namespace = {
+    name: '园岭小学',
+    url: 'https://ylxx.szftedu.cn/',
```
Comment (TonyRL) [inline lib/routes/szftedu/namespace.ts:5]:
```suggestion
    url: 'ylxx.szftedu.cn',
```

File: lib/routes/szftedu/dongtai.ts [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+            link: $('a', el).attr('href'),
+            pubDate: timezone(parseDate($('span[class=canedit]', el).text()), 8),
+        }));
+
+    const items = await Promise.all(
+        lists.map((item) =>
+            cache.tryGet(item.link, async () => {
+                const thisUrl = item.link;
+                const trueLink = thisUrl.includes("http") ? thisUrl : base_link + thisUrl;
+                const response = await got.get(trueLink);
```
Comment (TonyRL) [inline lib/routes/szftedu/dongtai.ts:50]: Please use either https://github.com/DIYgod/RSSHub/blob/a34ef4d7cf896fe08521edd2f474d6ac9509d61d/lib/routes/szftedu/dongtai.ts#L33 or https://github.com/DIYgod/RSSHub/blob/a34ef4d7cf896fe08521edd2f474d6ac9509d61d/lib/routes/szftedu/dongtai.ts#L50
But not both style

File: lib/routes/szftedu/dongtai.ts (hunk 4 above)
Comment (valuex) [reply lib/routes/szftedu/dongtai.ts]: suggestion committed.

File: lib/routes/szftedu/dongtai.ts (hunk 5 above)
Comment (valuex) [reply lib/routes/szftedu/dongtai.ts]: suggestion committed.

File: lib/routes/szftedu/gonggao.ts (hunk 7 above)
Comment (valuex) [reply lib/routes/szftedu/gonggao.ts]: suggestion committed.

File: lib/routes/szftedu/dongtai.ts [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+        supportScihub: false,
+    },
+    maintainers: ['valuex'],
+    handler,
+    description: '',
+};
+
+async function handler() {
+    const link = host;
+    const base_link = 'https://ylxx.szftedu.cn';
```
Comment (TonyRL) [inline lib/routes/szftedu/dongtai.ts:30]: > `camelCase`

https://github.com/DIYgod/RSSHub/pull/16876#discussion_r1773604909

File: lib/routes/szftedu/gonggao.ts [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+        }));
+
+    const items = await Promise.all(
+        lists.map((item) =>
+            cache.tryGet(item.link, async () => {
+                const thisUrl = item.link;
+                const trueLink = thisUrl.includes("http") ? thisUrl : host + thisUrl.substring(1);
+                const response = await got(trueLink);
+                const $ = load(response.data);
+                item.description = thisUrl.includes("http") ? $('#page-content').html() : $('div.pagenews04').html();
```
Comment (TonyRL) [inline lib/routes/szftedu/gonggao.ts:49]:
```suggestion
                item.description = thisUrl.includes("http") ? $('#page-content').html() : $('div.TRS_Editor').html();
```
`div.pagenews04` includes `title` and `pubDate` which have been set in their specific field. No need to include them again.

File: lib/routes/szftedu/dongtai.ts [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+        }));
+
+    const items = await Promise.all(
+        lists.map((item) =>
+            cache.tryGet(item.link, async () => {
+                const thisUrl = item.link;
+                const trueLink = thisUrl.includes("http") ? thisUrl : base_link + thisUrl;
+                const response = await got(trueLink);
+                const $ = load(response.data);
+                item.description = thisUrl.includes("http") ? $('#page-content').html() : $('div.pagenews04').html();
```
Comment (TonyRL) [inline lib/routes/szftedu/dongtai.ts:49]:
```suggestion
                item.description = thisUrl.includes("http") ? $('#page-content').html() : $('div.TRS_Editor').html();
```

**Pull request merged**